nixos/modules/services/web-apps/outline.nix

```
@@ -287,6 +287,43 @@ in
       );
     };
 
+    discordAuthentication = lib.mkOption {
+      description = ''
+        To configure Discord auth, you'll need to create an application at
+        https://discord.com/developers/applications/
+
+        See https://docs.getoutline.com/s/hosting/doc/discord-g4JdWFFub6
+        for details on setting up your Discord app.
+      '';
+      default = null;
+      type = lib.types.nullOr (
+        lib.types.submodule {
+          options = {
+            clientId = lib.mkOption {
+              type = lib.types.str;
+              description = "Authentication client identifier.";
+            };
+            clientSecretFile = lib.mkOption {
+              type = lib.types.str;
+              description = "File path containing the authentication secret.";
+            };
+            serverId = lib.mkOption {
+              type = lib.types.str;
+              description = ''
+                Restrict logins to a specific server (optional, but recommended).
+                You can find a Discord server's  by right-clicking the server icon,
+                and select “Copy Server ID”.
+              '';
+            };
+            serverRoles = lib.mkOption {
+              type = lib.types.str;
```

If it's supposed to be comma-separated, you can use `lib.types.commas`
```suggestion
              type = lib.types.commas;
```
which also will be mergeable, if defined in multiple places.